src/transformers/modeling_bart.py

```diff
@@ -0,0 +1,1052 @@
+}
+
+BART_START_DOCSTRING = r"""
+ "BART is a sequence to sequence model which uses a standard Transformer based Translation architecture.
+
+    This model is a PyTorch `torch.nn.Module`_ sub-class. Use it as a regular PyTorch Module and
+    refer to the PyTorch documentation for all matter related to general usage and behavior.
+
+    .. _`Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer`:
+        https://arxiv.org/abs/1910.10683
+
+    .. _`torch.nn.Module`:
+        https://pytorch.org/docs/stable/nn.html#module
+ Paper: BART: Denoising Sequence-to-Sequence Pre-training for Natural Language Generation, Translation, and Comprehension
+    https://arxiv.org/abs/1910.13461
+ Authors: Mike Lewis, Yinhan Liu, Naman Goyal, Marjan Ghazvininejad, Abdelrahman Mohamed, Omer Levy, Ves Stoyanov, Luke Zettlemoyer
+ (Submitted on 29 Oct 2019)
+ Code Ported from https://github.com/pytorch/fairseq/tree/master/examples/bart
+    An encoder decoder transformer pre-trained in a text-to-text denoising generative setting.
+
+    Parameters:
+        config (:class:`~transformers.BartConfig`): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the configuration.
+            Check out the :meth:`~transformers.PreTrainedModel.from_pretrained` method to load the model weights.
+"""
+
+BART_INPUTS_DOCSTRING = r"""
+    Inputs:
+        **input_ids**: ``torch.LongTensor`` of shape ``(batch_size, sequence_length)``:
+            Indices of input sequence tokens in the vocabulary. Use BartTokenizer.encode to produce them.
+            Padding will be ignored by default should you provide it.
+            Indices can be obtained using :class:`transformers.BartTokenizer.encode(text)`.
+            Also see :func:`transformers.PreTrainedTokenizer.convert_tokens_to_ids` for details.
+        **attention_mask**: (`optional`) ``torch.FloatTensor`` of shape ``(batch_size, sequence_length)``:
+            Mask to avoid performing attention on padding token indices in the encoder inputs.
+            Default: a mask will be created that ignore config.pad_token_id
+            Mask values selected in ``[0, 1]``:
+            ``1`` for tokens that are NOT MASKED, ``0`` for MASKED tokens.
+        **decoder_input_ids**: (`optional`) ``torch.FloatTensor`` of shape ``(batch_size, sequence_length)``:
+            only use for translation and summarization. Otherwise use the default which shifts the encoder's
+            input_ids right
+        **decoder_attention_mask**  `optional`) ``torch.FloatTensor`` of shape ``(batch_size, sequence_length)``:
+           default behavior ignore pad tokens and future tokens.
+             See diagram 1 in the paper for more info on the default strategy
+
+    read `prepare_bart_inputs` for more information on the default behavior.
```

Comment:
We've switched to a more uniform format, where you would have to 
1) rename the section from "Inputs" to "Args" for readthedocs/sphinx (our doc generator) to understand it 
2) link to the [glossary](https://huggingface.co/transformers/glossary.html) 
3) If possible re-use as similar docstrings as possible to the other models. Using different docstrings with different vocabulary is bound to confuse users.
4) (Optional) the [glossary](https://huggingface.co/transformers/glossary.html) currently doesn't contain any information related to the seq2seq models. It would be great if there was, but it is a lengthy process so it might be a better idea to do it once BART is wrapped up. Let me know if this is something that would be interesting for you.


You can check an example in the [BERT file](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bert.py#L545-L587).